core/src/subgraph/context/instance.rs

```diff
@@ -89,34 +90,31 @@ where
             // `SubgraphInstance::add_dynamic_data_source`. Ideally this should be refactored into
             // `IndexingContext`.
 
-            let runtime = ds.runtime();
-            let module_bytes = match runtime {
-                None => continue,
-                Some(ref module_bytes) => module_bytes,
-            };
-
             if let DataSource::Offchain(ds) = &ds {
                 // monitor data source only if it's not processed.
                 if !ds.is_processed() {
                     offchain_monitor.add_source(ds.source.clone())?;
                 }
             }
 
-            let host = this.new_host(logger.cheap_clone(), ds, module_bytes)?;
-            this.hosts.push(Arc::new(host));
+            let Some(host) = this.new_host(logger.cheap_clone(), ds)? else { continue };
+            this.hosts.push(host);
```

Comment:
I think this would be cleaner as a `if let Some(..)` so we can avoid the `continue`

Reply:
This code changes in a further commit.